### PR TITLE
DYN-5931 Lucene Search typo tolerance changes

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Per [DYN-5931](https://jira.autodesk.com/browse/DYN-5931), making changes to where all Lucene settings are and trying to modify our fuzzy search param. But from my own testing, it seems the FuzzyMinSim is always 2 (max edits) regardless of the float value. 

More to read at https://stackoverflow.com/questions/18317546/changing-similarity-in-lucene-fuzzyquery and https://blog.mikemccandless.com/2011/03/lucenes-fuzzyquery-is-100-times-faster.html

- When the search term length is less than 4, no typo is allowed
- When the search term length is between 5 to 7, typo with one edit is allowed
- When the search term length is larger than or equal to 8, typo with two edits are allowed

Example:
image
image

If the max edits are larger than 2, in the following case, the max edits would be 3, then no results would be shown. This matches what the author blog post wrote but I failed to find a way to make the max edits value less than 2
image

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCore/Configuration/LuceneConfig.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        /// Specify the Lucene.Net compatibility version
+        /// </summary>
+        public static LuceneVersion LuceneNetVersion = LuceneVersion.LUCENE_48;
+
+        /// <summary>
+        /// Default operator for Lucence query parser
+        /// </summary>
+        internal static Operator DefaultOperator = Operator.OR;
+
+        internal static float MinimumSimilarity = 0.5f;
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/LuceneConfig.cs:21]: @zeusongit When you started the change, did you try other values? I tried other values and it seems the setting does not make a diff, all the similarity threshold is 2

Comment (QilongTang) [reply src/DynamoCore/Configuration/LuceneConfig.cs:24]: oh nvm, I figured it out, I think the max edits are the ones in effect now

**Comment by QilongTang** [comment]
The reported regression passed locally

**Comment by QilongTang** [comment]
The next job reporting only one sporadic failure: [DynamoCoreWpfTests.CoreUserInterfaceTests.WorkspaceContextMenu_IfSubmenuOpenOnMouseHover](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/9384/testReport/junit/DynamoCoreWpfTests/CoreUserInterfaceTests/WorkspaceContextMenu_IfSubmenuOpenOnMouseHover/)

**Review by RobertGlobant20 — APPROVED** [review]
LGTM

**Review by reddyashish — APPROVED** [review]

**Pull request merged**